Src/VimCore/CoreInterfaces.fs

```diff
@@ -4606,6 +4606,90 @@ type HistoryList () =
             let seq = _list :> string seq
             seq.GetEnumerator()
 
+/// Represents a single line of text being edited
+[<StructuralEquality>]
+[<NoComparison>]
+[<Struct>]
+[<DebuggerDisplay("{ToString(),nq}")>]
+type EditableCommand =
+
+    val private _text: string
+    val private _caretPosition: int
+
+    /// Constructor for initial text with the caret at the end
+    new (text: string) =
+        { _text = text; _caretPosition = text.Length }
+
+    /// Constructor for text and caret position
+    new (text: string, caretPosition: int) =
+        { _text = text; _caretPosition = caretPosition }

```

Comment:
Did you consider using an optional parameter to make it a single ctor?

Reply:
Hmm ... that may not work well with the C# calling code though.

---
In reply to: [302303313](https://github.com/VsVim/VsVim/pull/2684#discussion_r302303313) [](ancestors = 302303313)

Reply:
I also thought about omitting the one argument constructor entirely so that each caller would be forced to at least consider whether the caret position was correct. I also agonized over the name of the type.  These are type names I seriously considered (seriously!):

- `CommandLineState`
- `CommandData` (already means something else)
- `CommandLineData`
- `CommandView` (analogous to `TextView`)
- `InProgressCommand`
- `EditCommand` (is it a command for editing or a a command line related to editing?)
- `CommandInput`
- `TextWithCaret`
- `EditableCommand`

In the end `EditableCommand` was the most descriptive and the least offensive!